drivers/led/issi/is31fl3761.c

```diff
@@ -0,0 +1,299 @@
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "is31fl3761.h"
+#include "i2c_master.h"
+#include "gpio.h"
+#include "wait.h"
+
+#include "rgb_matrix.h"
+
+/** 保证数据结构对齐到3761寄存器 */
+#define IS31FL3761_PWM_0_REGISTER_COUNT 252
+#define IS31FL3761_PWM_1_REGISTER_COUNT 252
+#define IS31FL3761_PWM_2_REGISTER_COUNT 90
+#define IS31FL3761_SCALING_REGISTER_COUNT 66
+#define IS31FL3761_SCALING_START_ADDR 0x5B
+
+#ifndef IS31FL3761_I2C_TIMEOUT
+#    define IS31FL3761_I2C_TIMEOUT 100
+#endif
+
+#ifndef IS31FL3761_I2C_PERSISTENCE
+#    define IS31FL3761_I2C_PERSISTENCE 0
+#endif
+
+#ifndef IS31FL3761_CONFIGURATION
+#    define IS31FL3761_CONFIGURATION 0x01
+#endif
+
+#ifndef IS31FL3761_PWM_FREQUENCY
+#    define IS31FL3761_PWM_FREQUENCY IS31FL3761_PWM_FREQUENCY_92K_HZ
+#endif
+
+#ifndef IS31FL3761_SW_PULLUP
+#    define IS31FL3761_SW_PULLUP IS31FL3761_PUR_16K_OHM
+#endif
+
+#ifndef IS31FL3761_CS_PULLDOWN
+#    define IS31FL3761_CS_PULLDOWN IS31FL3761_PDR_0K5_OHM
+#endif
+
+#ifndef IS31FL3761_GLOBAL_CURRENT
+#    define IS31FL3761_GLOBAL_CURRENT 0xFF
+#endif
+
+const uint8_t i2c_addresses[IS31FL3761_DRIVER_COUNT] = {
+    IS31FL3761_I2C_ADDRESS_1,
+#ifdef IS31FL3761_I2C_ADDRESS_2
+    IS31FL3761_I2C_ADDRESS_2,
+#    ifdef IS31FL3761_I2C_ADDRESS_3
+    IS31FL3761_I2C_ADDRESS_3,
+#        ifdef IS31FL3761_I2C_ADDRESS_4
+    IS31FL3761_I2C_ADDRESS_4,
+#        endif
+#    endif
+#endif
+};
+
+// These buffers match the IS31FL3761 and IS31FL3761A PWM registers.
+// The scaling buffers match the page 2 and 3 LED On/Off registers.
+// Storing them like this is optimal for I2C transfers to the registers.
+// We could optimize this and take out the unused registers from these
+// buffers and the transfers in is31fl3761_write_pwm_buffer() but it's
+// probably not worth the extra complexity.
+typedef struct is31fl3761_driver_t {
+    uint8_t pwm_buffer_0[IS31FL3761_PWM_0_REGISTER_COUNT];
+    uint8_t pwm_buffer_1[IS31FL3761_PWM_1_REGISTER_COUNT];
+    uint8_t pwm_buffer_2[IS31FL3761_PWM_2_REGISTER_COUNT];
+    bool    pwm_buffer_dirty;
+    uint8_t scaling_buffer[IS31FL3761_SCALING_REGISTER_COUNT];
+    bool    scaling_buffer_dirty;
+} PACKED is31fl3761_driver_t;
+
+is31fl3761_driver_t driver_buffers[IS31FL3761_DRIVER_COUNT] = {{
+    .pwm_buffer_0         = {0},
+    .pwm_buffer_1         = {0},
+    .pwm_buffer_2         = {0},
+    .pwm_buffer_dirty     = false,
+    .scaling_buffer       = {0},
+    .scaling_buffer_dirty = false,
+}};
```

Comment:
```suggestion
static is31fl3761_driver_t driver_buffers[IS31FL3761_DRIVER_COUNT];
```
The missing `static` is unfortunately duplicated in all LED drivers.  And the explicit initialization should not be needed (the static variable without an explicit initializer would be placed into the BSS section, where it would be zeroed during startup, and won't require any extra space in the firmware).